src/adapters/ip_camera_adapter/mod.rs

```diff
@@ -261,11 +262,22 @@ impl<T: Controller> Service for IpCameraService<T> {
             "snapshot" => self.cmd_snapshot(),
             "list" => self.cmd_list(),
             "get" => {
-                if let Some(filename) = req.url.query.clone() {
-                    // For now we assume that the query is the filename
-                    return self.cmd_get(&filename);
+                if let Some(query) = req.url.query.as_ref() {
+                    // Creating a fake URL to get the query parsed.
```

Comment:
Here I use the same hack as used in [1] to avoid adding custom query string processing code or one more dependency [2].

Please, tell me if you'd rather prefer adding [2] that has dedicated query string parsing capabilities.

[1] https://github.com/fxbox/foxbox/blob/996e84123e464eb69b36b9d960211c7b66150461/src/ws_server.rs#L48
[2] https://crates.io/crates/urlencoded